### ISSUE TITLE
Support nesting ROW in MULTISET

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12116]
A known limitation in jOOQ 3.15.0's `MULTISET` support was that `ROW` cannot be nested in `MULTISET` yet (see also https://github.com/jOOQ/jOOQ/issues/12021). It should be possible to run a query like this:

```java
Result<Record3<
    Integer,
    Record2<String, String>,
    Result<Record2<
        Record2<Integer, String>,
        Result<Record1<String>>
    >>
>> result = ctx.select(
    TAuthor_ID(),
    // Normal nested row ("top level")
    row(
        TAuthor_FIRST_NAME(),
        TAuthor_LAST_NAME()
    ),
    multiset(
        select(
            // Multiset-nested row
            row(
                TBook_ID(),
                TBook_TITLE()
            ).as("book"),
            multiset(
                select(TBookToBookStore_BOOK_STORE_NAME())
                .from(TBookToBookStore())
                .where(TBookToBookStore_BOOK_ID().eq(TBook_ID()))
                .orderBy(TBookToBookStore_BOOK_STORE_NAME())
            ).as("multiset")
        )
        .from(TBook())
        .where(TBook_AUTHOR_ID().eq(TAuthor_ID()))
    ).as("multiset")
)
.from(TAuthor())
.orderBy(TAuthor_ID())
.fetch();
```

**Comment by lukaseder** [comment]
This is such a powerful combination of features, I'd rather publish this already in a 3.15.x patch release than only in 3.16.0. Backport issue: https://github.com/jOOQ/jOOQ/issues/12122

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.1 (https://github.com/jOOQ/jOOQ/issues/12122)
